app-frontend/src/app/components/lab/colormapModal/colormapModal.js

```diff
@@ -107,14 +108,31 @@ class ColormapModalController {
     }
 
     onPaste(text) {
-        const result = JSON.parse(text.replace(/'/g, '"'));
-        this.breakpoints = this.colorsToBreakpoints(result); this.selectMode = 'form';
-        this.redistributePixelValues();
-        this.$scope.$evalAsync();
+        try {
+            const result = JSON.parse(text.replace(/'/g, '"'));
+            this.breakpoints = this.colorsToBreakpoints(result);
+            this.selectMode = 'form';
+            this.redistributePixelValues();
+            this.$scope.$evalAsync();
+        } catch (e) {
+            this.pasteError = e.message;
+            this.$scope.$evalAsync();
+        }
     }
 
     colorsToBreakpoints(colors) {
-        return colors.map((color, index) => ({value: index, color: color}));
+        let colorRegex = /^#([0-9a-f]{3}){1,2}$/i;
```

Comment:
Nice "one or two of exactly three number" regex :sunglasses:

Reply:
had to remove that :( chrome doesn't like 3 digit hexes